ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/data/realtime/RealtimeConversationHelperTest.java

```diff
@@ -0,0 +1,360 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.verifyStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+
+import android.os.Handler;
+import android.text.TextUtils;
+import com.kayako.sdk.android.k5.common.activities.MessengerOpenTracker;
+import com.kayako.sdk.android.k5.core.KayakoLogHelper;
+import com.kayako.sdk.android.k5.kre.base.KreConnection;
+import com.kayako.sdk.android.k5.kre.base.KreConnectionFactory;
+import com.kayako.sdk.android.k5.kre.base.KreSubscription.OnSubscriptionListener;
+import com.kayako.sdk.android.k5.kre.base.credentials.KreCredentials;
+import com.kayako.sdk.android.k5.kre.base.kase.KreCaseSubscription;
+import com.kayako.sdk.android.k5.kre.data.Change;
+import com.kayako.sdk.android.k5.kre.helpers.MinimalClientTypingListener;
+import com.kayako.sdk.android.k5.kre.helpers.RawCaseChangeListener;
+import com.kayako.sdk.android.k5.kre.helpers.RawCasePostChangeListener;
+import com.kayako.sdk.android.k5.kre.helpers.RawUserOnCasePresenceListener;
+import com.kayako.sdk.android.k5.messenger.data.conversation.viewmodel.UserViewModel;
+import com.kayako.sdk.base.callback.ItemCallback;
+import com.kayako.sdk.error.ErrorCode;
+import com.kayako.sdk.error.KayakoException;
+import com.kayako.sdk.messenger.conversation.Conversation;
+import com.kayako.sdk.messenger.message.Message;
+import java.util.Collections;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.rules.ErrorCollector;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({
+        KreConnectionFactory.class,
+        TextUtils.class,
+        KayakoLogHelper.class,
+        KreStarterFactory.class,
+        MessengerOpenTracker.class,
+        RealtimeConversationHelper.class,
+        LoadResourceHelper.class,
+        Handler.class
+})
+public class RealtimeConversationHelperTest {
+
+    private static final long USER_ID = 1L;
+    private static final long CONVERSATION_ID = 1L;
+    private static final int ENTRY_TIME = 0;
+    private static final long MESSAGE_ID = 1L;
+    private static final String USER_AVATAR = "userAvatar";
+    private static final String USER_NAME = "userName";
+    private static final String CHANNEL_NAME = "channelName";
+    private static final boolean IS_TYPING = true;
+    private static final String ERROR_MESSAGE = "exceptionMessage";
```

Comment:
✅ Good to see constant used instead of hard-coded values.

This approach increases maintainability (single point of modification) and debugging, avoid duplication and decreases verbosity.

**No change suggested, looks good!**	  

<a id="29529510-4ee7-11e8-b617-bd6867d59d97" href="none" codereview-uuid-reserved-tag />